storage/netapp_e/na_eseries_facts.py

```diff
@@ -0,0 +1,188 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+DOCUMENTATION = '''
+module: na_eseries_facts
+version_added: '2.2'
+short_description: Get facts about NetApp E-Series arrays
+extends_documentation_fragment:
+        - auth_basic
+options:
+  ssid:
+    required: true
+    description:
+    - The ID of the array to manage. This value must be unique for each array.
+
+description:
+    - Return various information about NetApp E-Series storage arrays (eg, configuration, disks)
+
+author: Kevin Hulquest (@hulquest)
+'''
+
+EXAMPLES = """
+---
+    - name: Get array facts
+      na_eseries_facts:
+        array_id: "{{ netapp_array_id }}"
+        api_url: "{{ netapp_api_url }}"
+        api_username: "{{ netapp_api_username }}"
+        api_password: "{{ netapp_api_password }}"
+        validate_certs: "{{ netapp_api_validate_certs }}"
+"""
+
+RETURN = """
+msg: Gathered facts for <StorageArrayId>.
+"""
+import json
+
+import os
+
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.basic import AnsibleModule, get_exception
+from ansible.module_utils.urls import open_url
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+
+
+def request(url, data=None, headers=None, method='GET', use_proxy=True,
+            force=False, last_mod_time=None, timeout=10, validate_certs=True,
+            url_username=None, url_password=None, http_agent=None, force_basic_auth=True, ignore_errors=False):
+    try:
+        r = open_url(url=url, data=data, headers=headers, method=method, use_proxy=use_proxy,
+                     force=force, last_mod_time=last_mod_time, timeout=timeout, validate_certs=validate_certs,
+                     url_username=url_username, url_password=url_password, http_agent=http_agent,
+                     force_basic_auth=force_basic_auth)
+    except HTTPError:
+        err = get_exception()
+        r = err.fp
+
+    try:
+        raw_data = r.read()
+        if raw_data:
+            data = json.loads(raw_data)
+        else:
+            data = None
+    except:
+        if ignore_errors:
+            pass
+        else:
+            raise
+
+    resp_code = r.getcode()
+
+    if resp_code >= 400 and not ignore_errors:
+        raise Exception(resp_code, data)
+    else:
+        return resp_code, data
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(
+        api_username=dict(type='str', required=True),
+        api_password=dict(type='str', required=True, no_log=True),
+        api_url=dict(type='str', required=True),
```

Comment:
Since we're making these required=true now, we also need to document them as such.  Talked to bcoca and unfortunately there's no current way to override documentation fragments so you'll have to remove the extends_documentation-fragment from the DOCUMENTATION section, copy the text into the modules documentation and then modify it to show that they are now required parameters.